main.py

```diff
@@ -19,23 +19,18 @@
 #     movingRainbow = movingRainbow+1
 
 ############################################################################################
-import sys
-print(sys.version)
-import time
 from digitalio import DigitalInOut, Direction, Pull
 import board
 import neopixel
 from rainbowio import colorwheel
+from light import Light
+import patterns
+from robot_states import recorded_mode, get_mode, get_states, ModeChangedException, frame_time_per_interval
+import threading
```

Comment:
⭐ ⭐ ⭐ try it on the real board, but I _thought_ micropython didn't support threading

Reply:
indeed it doesn't support threading, is there any other options we have that are similar to multithreading where multiple pieces of code can run at the same time without interrupting each other?